eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/model/DbsnpVariantAlleles.java

```diff
@@ -240,4 +265,37 @@ private String reverseComplementMicrosatelliteUnit(String str) {
             return reverseComplement(str);
         }
     }
+
+    /**
+     * This method unroll all the microsatellite 'units' in an allele. Each unit is a sequence motif and the number of
+     * times it is repeated, i.e. the microsatellite "(AG)5(C)4" has the units "(AG)5" and "(C)4". If expressed with the
+     * compressed syntax, alleles are also unrolled, e.g. (T)4 becomes TTTT.
+     *
+     * @param allelesArray Array containing all alleles
+     * @return Array containing all the unrolled alleles
+     */
+    private String[] unrollMicrosatelliteAlleles(String[] allelesArray) {
+        for (int i = 0; i<allelesArray.length; i++) {
+            StringBuilder unrolledAllele = new StringBuilder();
+            Matcher matcher = ANY_UNIT_PATTERN.matcher(allelesArray[i]);
+
+            while (matcher.find()) {
+                if (matcher.group(STR_MOTIF_REGEX_GROUP_NAME) != null) {
+                    // If a motif is detected, append it 'count' times
+                    String motif = matcher.group(STR_MOTIF_REGEX_GROUP_NAME);
```

Comment:
Would String motif = matcher.group(STR_MOTIF_REGEX_GROUP_NAME); and a subsequent if(motif != null) work?

Reply:
Yes, will do.